packages/sync-actions/test/shipping-methods.spec.js

```diff
@@ -0,0 +1,484 @@
+      }
+
+      const actual = shippingMethodsSync.buildActions(now, before)
+      const expected = [
+        { action: 'removeZone', zone: before.zoneRates[1].zone },
+      ]
+      expect(actual).toEqual(expected)
+    })
+
+    it('should build `removeZone` removing all existing zones', () => {
+      const before = {
+        zoneRates: [
+          { zone: { typeId: 'zone', id: 'z1' } },
+          { zone: { typeId: 'zone', id: 'z2' } },
+          { zone: { typeId: 'zone', id: 'z3' } },
+        ],
+      }
+      const now = {
+        zoneRates: [],
+      }
+
+      const actual = shippingMethodsSync.buildActions(now, before)
+      const expected = [
+        { action: 'removeZone', zone: before.zoneRates[0].zone },
+        { action: 'removeZone', zone: before.zoneRates[1].zone },
+        { action: 'removeZone', zone: before.zoneRates[2].zone },
+      ]
+      expect(actual).toEqual(expected)
+    })
+  })
+
+  describe('`addShippingRate`', () => {
+    it('should build `addShippingRate` action with one shipping rate', () => {
+      const before = {
+        zoneRates: [{ zone: { typeId: 'zone', id: 'z1' } }],
+      }
+      const now = {
+        zoneRates: [
+          {
+            zone: { typeId: 'zone', id: 'z1' },
+            shippingRates: [
+              { price: { centAmount: 1000, currencyCode: 'EUR' } },
+            ],
+          },
+        ],
+      }
+
+      const actual = shippingMethodsSync.buildActions(now, before)
+      const expected = [
+        {
+          action: 'addShippingRate',
+          zone: before.zoneRates[0].zone,
+          shippingRate: now.zoneRates[0].shippingRates[0],
+        },
+      ]
+      expect(actual).toEqual(expected)
+    })
+
+    it('should build `addShippingRate` action with multiple shipping rate', () => {
+      const before = {
+        zoneRates: [{ zone: { typeId: 'zone', id: 'z1' }, shippingRates: [] }],
+      }
+      const now = {
+        zoneRates: [
+          {
+            zone: { typeId: 'zone', id: 'z1' },
+            shippingRates: [
+              { price: { centAmount: 1000, currencyCode: 'EUR' } },
+              { price: { centAmount: 1000, currencyCode: 'USD' } },
+            ],
+          },
+        ],
+      }
+
+      const actual = shippingMethodsSync.buildActions(now, before)
+      const expected = [
+        {
+          action: 'addShippingRate',
+          zone: before.zoneRates[0].zone,
+          shippingRate: now.zoneRates[0].shippingRates[0],
+        },
+        {
+          action: 'addShippingRate',
+          zone: before.zoneRates[0].zone,
+          shippingRate: now.zoneRates[0].shippingRates[1],
+        },
+      ]
+      expect(actual).toEqual(expected)
+    })
+  })
+
+  describe('`removeShippingRate`', () => {
+    it('should build `removeShippingRate` removing one shippingRate', () => {
+      const before = {
+        zoneRates: [
+          {
+            zone: { typeId: 'zone', id: 'z1' },
```

Comment:
This is where your test is flawed. The zone can be expanded or is not. Your test assume that a consumer of the library always provides a non-expanded zone which is wrong I believe.